flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/PatternStream.scala

```diff
@@ -447,17 +447,17 @@ class PatternStream[T](jPatternStream: JPatternStream[T]) {
   }
 
  def sideOutputLateData(lateDataOutputTag: OutputTag[T]): PatternStream[T] = {
-   jPatternStream.sideOutputLateData(lateDataOutputTag)
+   jPatternStream = jPatternStream.sideOutputLateData(lateDataOutputTag)
```

Comment:
I'd rather suggest creating a new scala `PatternStream`. Having a mutable `var` does not look good.

Reply:
I think so, the best way is to rewrite PatternStream classs. Put PatternStream[T] builder into object PatternStream.

Reply:
Thank you for your valuable suggestions. I will try to rewrite it in a more elegant way.

Reply:
> I'd rather suggest creating a new scala `PatternStream`. Having a mutable `var` does not look good.

sir,  I have committed  a new version ,  is this way suitable？

Reply:
> I think so, the best way is to rewrite PatternStream classs. Put PatternStream[T] builder into object PatternStream.

thanks for your suggestion, but I don’t want to change too much code ，I think I need to exercise for a few more years

Reply:
> I think so, the best way is to rewrite PatternStream classs. Put PatternStream[T] builder into object PatternStream.

sir, thanks for your suggestion, but I don’t want to change too much code ，I think I still need to exercise for a few more years ...